components/movementsensor/rtkutils/cachedData.go

```diff
@@ -174,3 +180,49 @@ func (g *CachedData) Properties(
 		CompassHeadingSupported: true,
 	}, nil
 }
+
+// findDistance calculates the distance between two points on Earth.
+// lat1, lon1: Latitude and Longitude of point 1 (in decimal degrees)
+// lat2, lon2: Latitude and Longitude of point 2 (in decimal degrees)
+// returns the distance in meters.
+func findDistance(lat1, lon1, lat2, lon2 float64) float64 {
+	lat1Rad, lon1Rad := utils.DegToRad(lat1), utils.DegToRad(lon1)
+	lat2Rad, lon2Rad := utils.DegToRad(lat2), utils.DegToRad(lon2)
+
+	dLat := lat2Rad - lat1Rad
+	dLon := lon2Rad - lon1Rad
+
+	a := math.Sin(dLat/2)*math.Sin(dLat/2) +
+		math.Cos(lat1Rad)*math.Cos(lat2Rad)*
+			math.Sin(dLon/2)*math.Sin(dLon/2)
+	c := 2 * math.Atan2(math.Sqrt(a), math.Sqrt(1-a))
+
+	distanceKm := earthRadiusKm * c
+	return distanceKm * 1000 // convert km to meters
```

Comment:
Not sure if this is a useful tweak, but you might make the constant `earthRadiusM` so it's already in meters instead of kilometers, and then skip this line. Per [Wolfram Alpha](https://www.wolframalpha.com/input?i=earth+radius+in+meters),

```go
const earthRadiusM = 6371009 // Earth's radius in meters
```